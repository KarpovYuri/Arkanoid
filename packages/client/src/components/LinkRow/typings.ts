export type TLinkRowProps = {
  rowData: {
    cell1: string,
    cell2: string | number,
    cell3: string | number,
  },
  id: string,
  path: string
};
