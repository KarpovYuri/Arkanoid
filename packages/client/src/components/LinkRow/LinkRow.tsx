import { useNavigate } from 'react-router-dom';
import { TLinkRowProps } from './typings';

export function LinkRow({ rowData, path }: TLinkRowProps) {
  const navigate = useNavigate();
  const handleRowClick = ():void => {
    navigate(`./${path}`);
  };

  return (
    <tr onClick={handleRowClick}>
      <td>{rowData.cell1}</td>
      <td>{rowData.cell2}</td>
      <td>{rowData.cell3}</td>
    </tr>
  );
}
