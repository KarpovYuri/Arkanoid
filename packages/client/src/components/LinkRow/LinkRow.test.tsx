import { fireEvent, getByTestId, render, screen } from '@testing-library/react';
import { LinkRow } from './LinkRow';

const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

describe('LinkRow component tests', () => {
  const mockData = { cell1: 'first', cell2: 'second', cell3: 'third' };
  const mockPath = '/sample-test';

  test('Correct render', () => {
    render(<LinkRow rowData={mockData} path={mockPath} />);
    const component = screen.getByTestId('linkRow');

    expect(component).toMatchSnapshot();
  });

  test('should navigate, when LinkRow is clicked', () => {
    const tableBody = document.createElement('tbody');
    const { container } = render(
      <LinkRow rowData={mockData} path={mockPath} />,
      {
        container: document.body.appendChild(tableBody),
      },
    );
    const trow = getByTestId(container, 'linkRow');

    fireEvent.click(trow);
    expect(mockedUsedNavigate).toHaveBeenCalledTimes(1);
  });
});
