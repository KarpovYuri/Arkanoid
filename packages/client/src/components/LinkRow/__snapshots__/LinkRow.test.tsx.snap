// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkRow component tests Correct render 1`] = `
<tr
  data-testid="linkRow"
>
  <td>
    first
  </td>
  <td>
    second
  </td>
  <td>
    third
  </td>
</tr>
`;
